### ISSUE TITLE
AXE forces use of role when aria-label is used, contradicts the actual specs.

### ISSUE THREAD

**Issue by svdb99** [dequelabs/axe-core#4182]
### Product

axe-core

### Product Version

4.63.1

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

(AXE integrated in Jenkins automated build cycle. I don't have access to the Jenkins config so I'm not sure if this is **core** or **linter** specific.)

#### Expectation
The AXE rules should enforce the WAI-ARIA specifications, not contradict them.

#### Actual
When any element has an aria-label without a role attribute, the error thrown by AXE is `aria-label attribute cannot be used on a [insert element name] with no valid role attribute.`

This contradicts the actual specifications which don't require a role to be set at all.
[WAI-ARIA specifications for arial-label](https://www.w3.org/TR/wai-aria/#aria-label) 

The [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) even clarify with the following quote: 
"The aria-label attribute can be used with regular, semantic HTML elements; it is not limited to elements that have an ARIA role assigned."

#### How to Reproduce
`<div aria-label="hello world" ...>hello world</div>`

#### Context
As a best practice, it's understandable that the addition of a role attribute can sometimes clarify what an element is for, but that has to be decided on a case by case basis and can't be a general rule (which is why it isn't in the specs). When AXE blurs the lines between specs and best practice, it confuses people and causes even more mistakes. Unfortunately, the reality is that most developers rely on your tool so they wouldn't have to read the specs.

**Comment by straker** [comment]
Thanks for the issue. So ARIA has a concept of [prohibited states and properties](https://www.w3.org/TR/wai-aria/#prohibitedattributes) where ARIA attributes are not allowed to be used on certain roles. So for example if you look at the [caption role](https://www.w3.org/TR/wai-aria/#caption) it lists `aria-label` and `aria-labelledby` as prohibited states and properties so axe-core will flag an element with `role=caption` using `aria-label` as a violation.

A [div has an implicit role of generic](https://www.w3.org/TR/html-aria/#el-div) and the [generic role](https://www.w3.org/TR/wai-aria/#generic) prohibits using `aria-label`, `aria-labelledby`, and `aria-roledescription`, thus the violation. Additionally, screen readers do not announce the label on a generic element so the message you put there will be ignored by the audience it was intended for.

As for the Mozilla docs, they could probably use a clarification as not all semantic HTML elements are allowed to use `aria-label`, but in the general sense what it's trying to say is that you do not need to use `aria-label` combined with using the `role` attribute but could instead use it on a `button` element.
